File: c/function/project/scope/static-global/math.c
#include "math.h"

static int aAll = 1;
static int bAll = 2;    // 隐藏全局变量

int gAll = 123;
static int gAll1 = 456;

// 1. 函数名对全局(整个程序的各个文件.c)都是可见的 (不管有没有导入 .h 声明)
// 2. 限制函数在当前文件中使用: static
int max(int a, int b)
{
    return a > b ? a : b;
}

static int min(int a, int b)
{
    return a < b ? a : b;
}

File: cs/javascript/oop/build-in-object/Math.js
// property
console.log(Math.E); // e
console.log(Math.LN2); // LOGE 2
console.log(Math.LN10); // LOGE 10
console.log(Math.PI);
console.log(Math.SQRT1_2);
console.log(Math.SQRT2); // 2 的平方根

// method
console.log(Math.pow(1 / 2, 1 / 2) === Math.SQRT1_2);
console.log(Math.cbrt(1 / 8)); // 立方根
console.log(Math.log(10) === Math.LN10);

console.log(Math.abs(-10)); // absolute 绝对值
console.log(Math.ceil(9.01));
console.log(Math.floor(10.0000000000001));
console.log(Math.round(9.41)); // 四舍五入

console.log(Math.max(1, 2, 5, 6, 8, 3));
console.log(Math.min(1, 2, 5, 6, 8, 3));


// 伪随机数 (0,1)
console.log(Math.ceil(Math.random())); // 1
console.log(Math.floor(Math.random())); // 0


// Math -> instance
console.log(typeof Math); // object
console.log(Math instanceof Object); // true
console.log(Object.getPrototypeOf(Math) === Object.prototype); // true